src/filters/auth.rs

```diff
@@ -0,0 +1,30 @@
+//! HTTP Basic Authentication Filters
+
+use futures::future;
+use headers::{authorization::Basic, Authorization};
+
+use crate::filter::Filter;
+use crate::reject::Rejection;
+
+use super::header;
+
+/// Creates a `Filter` to the HTTP Basic Authentication header. If none was sent by the client, this filter will reject
+/// any request with a 401 Unauthorized.
+///
+/// # Example
+///
+/// ```
+/// use std::net::SocketAddr;
+/// use warp::Filter;
+/// use headers::{authorization::Basic, Authorization};
+///
+/// let route = warp::auth::basic("Realm name")
+///     .map(|auth_header: Authorization<Basic>| {
+///         println!("authorization header = {:?}", auth_header);
+///     });
+/// ```
+pub fn basic(
+    realm: &'static str,
+) -> impl Filter<Extract = (Authorization<Basic>,), Error = Rejection> + Copy {
+    header::header2().or_else(move |_| future::err(crate::reject::unauthorized(realm)))
```

Comment:
I feel like it would be nicer to have the callback argument be `Basic` instead of `Authorization`

Reply:
I don't follow what you mean here, can you please elaborate?

Reply:
Something like the following
```rust
pub fn basic(
    realm: &'static str,
) -> impl Filter<Extract = (Basic,), Error = Rejection> + Copy {
    header::header2().map(|auth: Authorization<Basic>| auth.0).or_else(move |_| future::err(crate::reject::unauthorized(realm)))
```

Reply:
thanks, okay I'll look into that!

Reply:
I see how that helps now, much cleaner code afterwards :) should be okay now too